src/home_robot_hw/home_robot_hw/nodes/state_estimator.py

```diff
@@ -86,44 +102,55 @@ def _odom_callback(self, pose: Odometry):
         pose_diff_log = (
             coeff * pose_diff_odom.log() + (1 - coeff) * pose_diff_slam.log()
         )
-        self._filtered_pose = self._filtered_pose * sp.SE3.exp(pose_diff_log)
+        self._filtered_pose = slam_pose * sp.SE3.exp(pose_diff_log)
```

Comment:
The [discrete high pass filter](https://en.wikipedia.org/wiki/High-pass_filter#Algorithmic_implementation) can be written as:
```py
coeff = dt / (RC + dt)  # RC = time constant
output[t] = coeff * (output[t-1] + (input[t] - input[t-1]))
```

The [discrete low pass filter](https://en.wikipedia.org/wiki/Low-pass_filter#Simple_infinite_impulse_response_filter) can be written as:
```py
coeff = dt / (RC + dt)  # RC = time constant
output[t] = coeff * output[t-1] + (1 - coeff) * input[t]
```

In the high pass filter case, we are injecting the output signal with the difference between measurements, while in the low pass filter case, we are injecting the output signal with the absolute value of the measurements. 

The part where I originally fused the two filters to process the signals is a bit more hand-wavy, but I simply added together the injected signals of both slam + LPF and odom + HPF into the output pose:
```py
output_pose[t] = coeff * output_pose[t-1] + (1 - coeff) * input_slam[t] + coeff * (input_odom[t] - input_odom[t-1])
```
which can be re-written as
```py
output_pose[t] = output_pose[t-1] + (1 - coeff) * (input_slam[t] - output_pose[t-1]) + coeff * (input_odom[t] - input_odom[t-1])
```

Happy to discuss the mathematically sound implementation, but I think that we definitely need to compound on the previous filtered pose to perform correct filtering because
1. It's mathematically correct
2. Else a large SLAM error spike will throw off the output, which we want to alleviate with a proper low pass filter

Reply:
let's talk about this today

Reply:
i think we decided this was ok